### DISCUSSION TITLE
Delaying Retries for Specific Tests

### DISCUSSION THREAD

**Discussion by Masood-Ahmed271** [cypress-io/cypress#27080] [Questions and Help]
I want to retry certain test cases but I want to retry them after say 2 minutes or 3 minutes and during that time frame of 2 to 3 minutes, I want other test cases to run. Is it possible to do so? Or in an other scenario, when I want to retry a specific test, can it be possible to retry the test after all other tests have been run, for example I want to retry test b, and d, and the current test cases that I have is a,b,c,d, and e so the execution should look like a (no-retry), b (first try), c(no-retry), d(first-try), e(no-retry), b(second-retry), d(second-retry).

I hope somebody can answer my query!

**Comment by portyu9** [comment]
Hi @Masood-Ahmed271 

Built-in Cypress test retries do not support that execution order.

A failed test is retried as part of the same test run; Cypress does not have an option like:

```text
a → b attempt 1 → c → d attempt 1 → e → b attempt 2 → d attempt 2
```

and putting a long `cy.wait()` into the retry would simply block that Cypress runner rather than allowing other tests to execute meanwhile.

For a delay of several minutes, the better design is to make it a **second CI run/pass**, after the rest of the suite has completed.

There is also a newer option that did not exist when this question was posted: Cypress Cloud now has **Re-run optimization**, and with Cypress 15.21.0+ it can re-run only the tests that did not pass in the previous recorded CI run. That is currently a Cypress Cloud Business/Enterprise feature and is separate from normal test retries.

So for your desired behavior, either orchestrate a second pass in CI (with whatever delay you need), or use Cloud's failed-test rerun feature if applicable.

If the 2–3 minute delay is actually waiting for eventual consistency in the application/backend, it is usually better to poll the specific condition instead of deliberately failing and retrying the entire test.

Please mark this answer as accepted if it helped, thank you.
